pkg/analyzer/connectivity/connectivity.go

```diff
@@ -18,26 +18,22 @@ func (c ConnMap) Add(src, dst topology.Endpoint, conn *DetailedConnection) {
 	c[src][dst] = conn
 }
 
-// InitPairs adds all possible pairs with allow-all or deny-all, based on initAllow
-func (c ConnMap) InitPairs(initAllow bool, vms []topology.Endpoint, vmsFilter []string) {
-	vmsToaAnalyze := map[string]bool{}
-	if len(vmsFilter) > 0 {
-		for _, vmName := range vmsFilter {
-			vmsToaAnalyze[vmName] = true
-		}
-	}
-	for _, src := range vms {
-		for _, dst := range vms {
+// InitPairs adds all possible pairs from/to endpoints1 to/from endpoints2, with allow-all or deny-all, based on initAllow
+func (c ConnMap) InitPairs(initAllow bool, endpoints1, endpoints2 []topology.Endpoint, vmsFilter []string) {
+	filterFunc := func(ep topology.Endpoint) bool { return len(vmsFilter) > 0 && slices.Contains(vmsFilter, ep.Name()) }
+	endpoints1 = slices.DeleteFunc(slices.Clone(endpoints1), filterFunc)
+	endpoints2 = slices.DeleteFunc(slices.Clone(endpoints2), filterFunc)
```

Comment:
- why deleting endpoints for which `filterFunc` returns true?
- does it keep ip-block endpoints?

Reply:
1. fixed
2. currenly not, will alaborate it when implementing proper user filter

Reply:
should open an issue to support (2)?

Reply:
I ment this issue:
https://github.com/np-guard/vmware-analyzer/issues/244

Reply:
ok, added a comment on this issue.